src/main/java/Duke.java

```diff
@@ -1,10 +1,109 @@
+            userInput = myObj.nextLine() + " ";
+            subUserInput = userInput.substring(0, userInput.indexOf(' '));
+            try{
+                markedValue = Integer.parseInt(userInput.substring(userInput.indexOf(' ')).replaceAll("\\s+","")) - 1;
+            }
+            catch(Exception ignored){
+
+            }
+
+            try{
+                commandType = Commands.valueOf(subUserInput.toUpperCase());
+                switch (commandType) {
+                    case LIST:
+                        System.out.println("____________________________________________________________");
+                        for (int i = 0; i < todoList.length; i++) {
+                            if (todoList[i] != null) {
+                                System.out.println("\t" + (i + 1) + ") [" + todoList[i].getStatusIcon() +"] " + todoList[i].getDescription());
+                            } else {
+                                break;
+                            }
+                        }
+                        System.out.println("____________________________________________________________");
+                        break;
+
+                    case BYE :
+                        bye = true;
+                        break;
+
+                    case MARK :
+                        todoList[markedValue].markAsDone();
+                        System.out.println("____________________________________________________________");
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.println("[" +todoList[markedValue].getStatusIcon() +"]  " + todoList[markedValue].getDescription());
+                        System.out.println("____________________________________________________________");
+                        break;
+
+                    case UNMARK :
+                        todoList[markedValue].unMark();
+                        System.out.println("____________________________________________________________");
+                        System.out.println("OK, I've marked this task as not done yet:");
+                        System.out.println("[" +todoList[markedValue].getStatusIcon() +"]  " + todoList[markedValue].getDescription());
+                        System.out.println("____________________________________________________________");
+                }
+            }
+            catch(Exception e){
+                for(int i = 0; i < todoList.length; i++)
+                    if(todoList[i] == null) {
+                        Task newTask = new Task(userInput);
+                        todoList[i] = newTask;
+                        break;
+                    }
+                System.out.println("____________________________________________________________");
+                System.out.println("\t Added: " + userInput);
+                System.out.println("____________________________________________________________");
+            }
```

Comment:
I like how you have already tried to deal with exceptions, anomalous input with the try and catch statements.